packages/web-components/src/drawer/drawer.ts

```diff
@@ -0,0 +1,222 @@
+ *
+ * @tag fluent-drawer
+ */
+export class Drawer extends FASTElement {
+  /**
+   * @public
+   * The connectedCallback method of the custom element
+   */
+  public connectedCallback(): void {
+    super.connectedCallback();
+    Updates.enqueue(() => {
+      this.validateConfiguration();
+    });
+  }
+
+  /**
+   * @public
+   * Determines whether the drawer should be displayed as modal, non-modal, or alert.
+   * When in modal or alert mode, an overlay is applied over the rest of the view.
+   */
+  @attr({ attribute: 'type' })
+  public type: DrawerType = DrawerType.modal;
+
+  /**
+   * @public
+   * The ID of the element that labels the drawer.
+   */
+  @attr({ attribute: 'aria-labelledby' })
+  public ariaLabelledby?: string;
+
+  /**
+   * @public
+   * The ID of the element that describes the drawer.
+   */
+  @attr({ attribute: 'aria-describedby' })
+  public ariaDescribedby?: string;
+
+  /**
+   * @public
+   * @defaultValue false
+   * Sets the drawer as inline
+   */
+  @attr({ mode: 'boolean' })
+  public inline?: boolean = false;
+
+  /**""
+   * @public
+   * @defaultValue start
+   * Sets the position of the drawer (start/end).
+   */
+  @attr
+  public position: DrawerPosition = DrawerPosition.start;
+
+  /**
+   * @public
+   * @defaultValue medium
+   * Sets the size of the drawer (small/medium/large).
+   */
+  @attr({ attribute: 'size' })
+  public size: DrawerSize = DrawerSize.medium;
+
+  /**
+   * @public
+   * The dialog element.
+   */
+  @observable
+  public dialog!: HTMLDialogElement;
+
+  /**
+   * @public
+   * Method to emit an event after the dialog's open state changes
+   * HTML spec proposal: https://github.com/whatwg/html/issues/9733
+   */
+  public emitToggle = (): void => {
+    this.$emit('toggle', {
+      oldState: this.dialog.open ? 'closed' : 'open',
+      newState: this.dialog.open ? 'open' : 'closed',
+    });
+  };
+
+  /**
+   * @public
+   * Method to emit an event after the dialog's open state changes
+   * HTML spec proposal: https://github.com/whatwg/html/issues/9733
+   */
+  public emitBeforeToggle = (): void => {
+    this.$emit('beforetoggle', {
+      oldState: this.dialog.open ? 'open' : 'closed',
+      newState: this.dialog.open ? 'closed' : 'open',
+    });
+  };
+
+  /**
+   * @public
+   * Method called when the 'inline' attribute changes
+   */
+  public inlineChanged(oldValue: boolean, newValue: boolean): void {
+    if (this.$fastController.isConnected) {
+      if (newValue) {
+        this.validateConfiguration();
+      }
+    }
+  }
+
+  /**
+   * @public
+   * Method called when the 'type' attribute changes
+   */
+  public typeChanged(oldValue: boolean, newValue: boolean): void {
+    if (this.$fastController.isConnected) {
+      if (newValue != oldValue) {
+        this.validateConfiguration();
+      }
+    }
+  }
```

Comment:
Seeing these two together doing the same job makes me think we shouldn't expose `type` at all and only expose `inline` as an attribute. Thoughts?